src/internal/pachd/enterprise.go

```diff
@@ -0,0 +1,71 @@
+package pachd
+
+import (
+	"context"
+	"path"
+
+	"google.golang.org/grpc"
+
+	"github.com/pachyderm/pachyderm/v2/src/enterprise"
+	eprsserver "github.com/pachyderm/pachyderm/v2/src/server/enterprise/server"
+)
+
+// An EnterpriseBuilder builds an enterprise-mode pachd.  It should only be
+// created with NewEnterpriseBuilder.
+//
+// Enterprise mode is the enterprise server which is used to manage multiple
+// Pachyderm installations.
+type EnterpriseBuilder struct {
+	builder
+}
+
+func (eb *EnterpriseBuilder) registerEnterpriseServer(ctx context.Context) error {
+	eb.enterpriseEnv = eprsserver.EnvFromServiceEnv(
+		eb.env,
+		path.Join(eb.env.Config().EtcdPrefix, eb.env.Config().EnterpriseEtcdPrefix),
+		eb.txnEnv,
+	)
+	apiServer, err := eprsserver.NewEnterpriseServer(
```

Comment:
Is it worth having a helper function that calls the statements from lines 28 - 40 which takes a `enterpriseEnv` as a parameter, this way the code for creating and registering the server is not repeated?